Makefile

```diff
@@ -97,4 +105,96 @@ unit-tests:
 
 #
 ##############################################################
-include terraform.mk
+### Docker local dev #########################################
+#
+oauth/pkcs12/certificate.pfx:
+	# All calls to the openssl cli happen in the oidc-server-mock container.
+	@echo "Generating certificates for local dev"
+	docker run -v $(PWD)/oauth/pkcs12:/tmp/certs --workdir /tmp/certs --rm=true --entrypoint bash soluto/oidc-server-mock:0.3.0 ./generate_cert.sh
+	@if [ "$$(uname -s)" == "Darwin" ]; then \
+		echo "Installing generated certs into the local keychain requires sudo access:"; \
+		sudo security add-trusted-cert -d -p ssl -k /Library/Keychains/System.keychain oauth/pkcs12/server.crt; \
+	fi
+	# Linux assumes Ubuntu
+	if [ "$$(uname -s)" == "Linux" ]; then \
+		sudo cp oauth/pkcs12/server.crt /usr/local/share/ca-certificates/; \
+		sudo update-ca-certificates; \
+	fi
+	docker run -v $(PWD)/oauth/pkcs12:/tmp/certs --workdir /tmp/certs --rm=true --entrypoint bash soluto/oidc-server-mock:0.3.0 ./generate_pfx.sh
+	# On Linux, the pkcs12 directory gets written to with root permission. Force ownership to our user.
+	sudo chown -R $$(id -u):$$(id -g) $(PWD)/oauth/pkcs12
+
+.env.ecr:
+	echo DOCKER_REPO=$$(aws sts get-caller-identity --profile $(AWS_DEV_PROFILE) | jq -r .Account).dkr.ecr.us-west-2.amazonaws.com/ > .env.ecr;
+
+.PHONY: local-ecr-login
+local-ecr-login:
+	if PROFILE=$$(aws configure list-profiles | grep $(AWS_DEV_PROFILE)); then \
+		aws ecr get-login-password --region us-west-2 --profile $(AWS_DEV_PROFILE) | docker login --username AWS --password-stdin $$(aws sts get-caller-identity --profile $(AWS_DEV_PROFILE) | jq -r .Account).dkr.ecr.us-west-2.amazonaws.com; \
+	fi
+
+.PHONY: local-init
+local-init: oauth/pkcs12/certificate.pfx .env.ecr local-ecr-login ## Launch a new local dev env and populate it with test data.
+	docker-compose $(COMPOSE_OPTS) up -d
+	# docker-compose exec -T utility pip3 install awscli
+	docker-compose exec -T utility $(BACKEND_APP_ROOT)/scripts/setup_dev_data.sh
+	docker-compose exec -T utility python scripts/setup_localdata.py
```

Comment:
why do this?  also, how does the database get written to before the `db --docker create` command?

Reply:
Docker-compose is launching the db container at `docker-compose up -d` and the image we're pulling is your snapshot from S3.